File: src/main/scala/edu/jhu/fcriscu1/als/graphdb/value/AlsAssociatedGene.scala
package org.nygenome.als.graphdb.value

import org.apache.commons.csv.CSVRecord

case class AlsAssociatedGene(
                            geneSymbol: String,
                            geneName: String
                            )  {
  val id:String = geneSymbol
}
object AlsAssociatedGene extends ValueTrait {

  def parseCSVRecord(record:CSVRecord):AlsAssociatedGene = {
    new AlsAssociatedGene(
      record.get("Gene symbol"),
      record.get("Gene name")
    )
  }
}

File: src/main/kotlin/org/biodatagraphdb/alsdb/model/AlsAssociatedGene.kt
package org.biodatagraphdb.alsdb.model

import org.apache.commons.csv.CSVRecord

/**
 * Created by fcriscuo on 2/3/20.
 */


data class AlsAssociatedGene (val geneSymbol: String, val geneName: String){
    val id = geneSymbol

    companion object : AlsdbModel {
        const val GENE_SYMBOL = "Gene symbol"
        const val GENE_NAME = "Gene name"
        fun parseCSVRecord(record: CSVRecord): AlsAssociatedGene =
                AlsAssociatedGene(record.get(GENE_SYMBOL), record.get(GENE_NAME))


    }
}